code/obj/item/device/pda2/pda2.dm

```diff
@@ -1026,11 +1026,8 @@
 			if(. && (src.r_tone?.overrideAlert || src.r_tone_temp?.overrideAlert))
 				alert_message = .
 
-			for (var/atom in mobs)
-				if (!atom) continue
-				var/mob/O = atom
-				if (get_dist(get_turf(src),O) <= 3)
-					O.show_message(text("[bicon(src)] *[alert_message]*"))
+			for (var/mob/M in hearers(3,get_turf(src)))
+				M.show_message(text("[bicon(src)] *[alert_message]*"))
```

Comment:
show_message actually supports hearing-only if you pass 2 for the second argument (no, nobody's made defines for it AFAIK), so you can get away with just a range check

Reply:
Then why not use that

Reply:
is that prefered to looping over hearers?

Reply:
yes lol

Reply:
what should i loop over to get the said list of people i want to show the sound to?

Reply:
all_hearers?

Reply:
audible_message()

Reply:
i did the requested change (tho this changes it from beeping in a radius of 3 to beeping to all people who can hear it period)

Reply:
fine by me